dts/arm/st/h7/stm32h7.dtsi

```diff
@@ -45,11 +45,12 @@
 		};
 	};
 
-	quadspi_memory: memory@90000000 {
-		compatible = "zephyr,memory-region", "mmio-sram";
-		reg = <0x90000000 DT_SIZE_M(256)>;
-		zephyr,memory-region = "QSPI";
-		zephyr,memory-attr = <( DT_MEM_ARM(ATTR_MPU_EXTMEM) )>;
+	ext_memory: memory@90000000 {
+		compatible = "zephyr,memory-region";
+		reg = <0x90000000 DT_SIZE_M(64)>; /* 512 Mbits */
+		zephyr,memory-region = "EXTMEM";
+		/* The ATTR_MPU_EXTMEM attribut causing a MPU FAULT */
```

Comment:
This will break https://github.com/zephyrproject-rtos/zephyr/pull/57467 which required a strongly ordered memory.
If this is incompatible, this means we should define 2 different targets ("variants" in hwmv2) which would define different `zephyr,memory-attr` depending if memory is expected to be used in XIP or storage

Reply:
The https://github.com/zephyrproject-rtos/zephyr/blob/main/include/zephyr/arch/arm/mpu/arm_mpu_v7m.h 
sets the REGION_EXTMEM_ATTR to be STRONGLY_ORDERED_SHAREABLE
And the Memory region shareability and cache policies  for the stm32h7 and stm32F7
defines the 0x80000000-0x9FFFFFFF  range Non-shareable Write-through:
Could this difference explain the pb ?

![Screenshot from 2024-02-21 17-58-32](https://github.com/zephyrproject-rtos/zephyr/assets/51417392/4e314f19-cedf-4df6-b454-35414beb24c7)

Reply:
The [AN4760]( https://www.st.com/resource/en/application_note/dm00227538.pdf) and [AN4861](https://www.st.com/resource/en/application_note/dm00287603.pdf) propose to define the entire region as strongly ordered, non-cacheable, no access, execute never (_while memory-mapped mode is not enabled_). The `ATTR_MPU_EXTMEM` more or less follows, and it is ok for non-xip targets.

However, with xip, the attributes should be modified or an extra xip region shall be defined inside the extmem region. That region should start at the mapped memory address with the size of the slot.

The xip region attributes might differ for bootloader (_region can stay as strongly ordered, non-cacheable, execute never, but require rw access_) and application (_region follows RM and is defined as normal, non-shareable, and require to execute and read with optional write access_).

Some configurations are not defined within current [arm_mpu_v7m](https://github.com/zephyrproject-rtos/zephyr/blob/main/include/zephyr/arch/arm/mpu/arm_mpu_v7m.h).

The [MPU usage in STM32 with ARM Cortex®-M7](https://drive.google.com/file/d/1g_-mDfAIYs99pRifeAvfpunHBVIdq_a8/view) is quite helpful.

Reply:
OK, keeping the ATTR_MPU_EXTMEM  for the entire region:
which can be quad-spi _or_ octo-spi depending on  the stm32h7x device.